src/main/java/org/codehaus/plexus/archiver/zip/AddedDirs.java

```diff
@@ -65,19 +65,12 @@ public void clear()
      */
     public boolean update( String vPath )
     {
-        if ( addedDirs.get( vPath ) != null )
-        {
-            // don't add directories we've already added.
-            // no warning if we try, it is harmless in and of itself
-            return true;
-        }
-        addedDirs.put( vPath, vPath );
-        return false;
+        return !addedDirs.add( vPath );
```

Comment:
Hmm, so the existing code returns `true` if `vPath` was already added, right? I ask because the JavaDoc says otherwise and wanted to be sure I'm not overlooking something.

Reply:
Good catch.  I can change the javadoc, I suppose it's the safest way out...